### PR TITLE
update documents.md

### PR DESCRIPTION
**Pull request by maryamsulemani97**
Part of https://github.com/meilisearch/documentation/issues/1333

This PR updates `/learn/core_concepts/documents.md`
I have removed the "[Limitation and requirements](https://docs.meilisearch.com/learn/core_concepts/documents.html#limitations-and-requirements)" section and moved the content to known _imitations.md

### CODE REVIEW

File: learn/core_concepts/documents.md [hunk 1]
```diff
@@ -1,26 +1,58 @@
...
-- **Attribute**: the first part of a field. Acts as a name or description for its associated value.
-- **Value**: the second part of a field, consisting of data of any valid JSON type
-- **[Primary Field][primary-field]**: A special field that is mandatory in all documents. It contains the primary key and document identifier.
-- **[Primary Key][primary-key]**: the attribute of the primary field. **All documents in the same index must possess the same primary key.** Its associated value is the document identifier.
-- **[Document Identifier][document-id]**: the value of the primary field. **Every document in a given index must have a unique identifier**.
+- **Document**: An object which contains data in the form of one or more fields
+- **[Field](#fields)**: A set of two data items that are linked together: an attribute and a value
+- **Attribute**: The first part of a field. Acts as a name or description for its associated value
+- **Value**: The second part of a field, consisting of data of any valid JSON type
+- **[Primary Field](#primary-field)**: A special field that is mandatory in all documents. It contains the primary key and document identifier
```
Comment (guimachiavelli) [inline learn/core_concepts/documents.md:15]: I thought we had agreed on capitalizing the first letter after a colon only when it was used in headings? Not sure I'm sold on capitalizing it on list items.

File: learn/core_concepts/documents.md [hunk 2]
```diff
@@ -1,26 +1,58 @@
...
+
+If a field contains an object, Meilisearch flattens it during indexation using dot notation and brings the object's keys and values to the root level of the document itself. This flattened object is an intermediary representation and you will get the original structure upon search. You can read more about this in our [dedicated guide](/learn/advanced/datatypes.md#objects).
+
+You can apply [ranking rules](/learn/core_concepts/relevancy.md#ranking-rules) to some fields. For example, you may decide recent movies should be more relevant than older ones. You can also configure how Meilisearch handles certain fields at an [index level](/learn/configuration/settings.md) in the settings.
+
+### Displayed and searchable fields
+
+By default, all fields in a document are both displayed and searchable. You can modify this behavior using the [update settings endpoint](/reference/api/settings.md#update-settings), or the respective update endpoints for [displayed attributes](/reference/api/displayed_attributes.md#update-displayed-attributes), and [searchable attributes](/reference/api/searchable_attributes.md#update-searchable-attributes) so that a field is:
```
Comment (guimachiavelli) [inline learn/core_concepts/documents.md:31]: Honest question: is it clear enough what we mean by displayed and searchable fields? Would users benefit from a sentence here defining these two "types" of fields?

File: learn/core_concepts/documents.md [hunk 3]
```diff
@@ -1,26 +1,58 @@
...
+**Meilisearch uses a lot of RAM when indexing documents**. Be aware of your [RAM availability](/resources/faq.md#what-are-the-recommended-requirements-for-hosting-a-meilisearch-instance) as you increase your batch size as this could cause Meilisearch to crash.
+
+When using the [add new documents endpoint](/reference/api/documents.md#add-or-update-documents), all documents must be sent in an array even if there is only one document.
 
 ### Dataset format
 
-You can provide your dataset in the following formats:
+Meilisearch accepts datasets in the following formats:
 
 - [JSON](#json)
```
Comment (guimachiavelli) [inline learn/core_concepts/documents.md:57]: Are these misaligned on the file itself or is it just an artefact of github's interface?

Comment (maryamsulemani97) [reply learn/core_concepts/documents.md:57]: It's GitHub, they're aligned:

Screen Shot 2022-06-07 at 12 15 09

**Review by guimachiavelli — APPROVED** [review]
🦀

**Comment by maryamsulemani97** [comment]
bors merge

**Pull request merged**